book/integration-service.md

```diff
@@ -0,0 +1,271 @@
+# Integration Service
+
+
+## Overview
+
+The Integration Service is composed of a couple controllers that facilitate automated testing of content produced by the HACBS build pipelines. It is responsible for integration testing capabilities.
+
+To support this capability, it needs to do a couple of things.
+
+- When a build pipeline completes, it creates an Snapshot CR representing a new collection of components that should be tested together.
+- When it sees a new Snapshot CR (created either by itself or by a user), it coordinates deployment of the application for testing. It does this by creating a new “ephemeral” StoneSoup Environment CRs which should in turn trigger the GitOps Service to provision environment (and a sub-workspace in CPS)
+- When it sees the environment is ready, it triggers Tekton PipelineRuns supplied by the user that run application tests and validation against the app running in that ephemeral Environment.
+- When it sees that all of the test PipelineRuns that it created have passed, it finishes its work by triggering a deployment of the application to the lowest known “static” environment defined by the user, usually staging.
```

Comment:
```After all test and validation PipelineRuns pass, the integration service will then trigger an application deployment to the lowest defined "static" environment (for example, staging).```

Reply:
Is it important/pivotal that the tests and validation are PipelineRuns? I really only see references to that in this list (and my suggestion on the previous item removed the reference). If this is important to highlight, I feel like we should be making a bigger call-out of it elsewhere.

Reply:
It is important because the tests and validations are executed by running Tekton PipelineRuns so it shapes how the tests are defined and configured by the user.

Reply:
Cool. If Tekton PipelineRuns are a pivotal part of the technology, I think we should at least call it out in the initial part of the overview.

Reply:
The updated overview should have Tekton mentioned in it